### PR TITLE
support limit for select

### PR DESCRIPTION
**Pull request by fqliao**

### CODE REVIEW

File: libstorage/MemoryTable2.cpp [hunk 1]
```diff
@@ -43,6 +43,37 @@ Entries::ConstPtr MemoryTable2::select(const std::string& key, Condition::Ptr co
     return selectNoLock(key, condition);
 }
 
+void MemoryTable2::proccessLimit(const Condition::Ptr& condition,
+    const std::shared_ptr<dev::storage::Entries>& entries, const Entries::Ptr& resultEntries)
```
Comment (bxq2011hust) [inline libstorage/MemoryTable2.cpp]: `void MemoryTable2::proccessLimit(const Condition::Ptr& condition, const Entries::Ptr& entries, const Entries::Ptr& resultEntries)`

File: libstorage/MemoryTable2.cpp [hunk 2]
```diff
@@ -43,6 +43,37 @@ Entries::ConstPtr MemoryTable2::select(const std::string& key, Condition::Ptr co
...
+{
+    if (condition->getOffset() >= 0 && condition->getCount() >= 0)
+    {
+        std::vector<size_t> limitedIndex;
+        limitedIndex.reserve(entries->size());
+        for (size_t i = 0; i < entries->size(); ++i)
```
Comment (bxq2011hust) [inline libstorage/MemoryTable2.cpp]: This can be simplify. you don't need index any more. Just add asked number of entry's point to entries.

Comment (fqliao) [reply libstorage/MemoryTable2.cpp]: recontruct the code

Comment (fqliao) [reply libstorage/MemoryTable2.cpp]: reconstruct the code

**Pull request merged**